src/write/line.rs

```diff
@@ -632,6 +645,20 @@ impl LineProgram {
                 if self.file_has_md5 {
                     w.write(&info.md5)?;
                 }
+                if self.file_has_source {
+                    // Note: An empty DW_LNCT_source is interpreted as missing
+                    // source code. Included source code should always be
+                    // terminated by a "\n" line ending.
+                    let empty_str = LineString::String(Vec::new());
+                    let source = info.source.as_ref().unwrap_or(&empty_str);
+                    source.write(
+                        w,
+                        constants::DW_FORM_string,
```

Comment:
Ideally users should be able to use `DW_FORM_line_strp` too, but that has the problem of needing to have added an empty string already, and doing that won't be necessary for DWARF 6. So for now, let's just document that only `LineString::String` is supported for `source`.

Reply:
Added a note to the `write::FileInfo` field. :+1: